### PR TITLE
DATAES-570 - Use Delete By Query API for delete by query operations.

### PR DESCRIPTION
**Pull request by xhaggi**
- [x] You have read the [Spring Data contribution guidelines](https://github.com/spring-projects/spring-data-build/blob/master/CONTRIBUTING.adoc).
- [x] There is a ticket in the bug tracker for the project in our [JIRA](https://jira.spring.io/browse/DATAES).
- [x] You use the code formatters provided [here](https://github.com/spring-projects/spring-data-build/tree/master/etc/ide) and have them applied to your changes. Don’t submit any formatting related changes.
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] You added yourself as author in the headers of the classes you touched. Amend the date range in the Apache license header if needed. For new types, add the license header (copy from another file and set the current year only).

### CODE REVIEW

**Comment by xhaggi** [comment]
This PR needs #279 to pass the tests successfully

**Comment by xhaggi** [comment]
@christophstrobl @sothawo would you mind taking a look?

File: src/main/java/org/springframework/data/elasticsearch/core/ElasticsearchRestTemplate.java [hunk 1]
```diff
@@ -863,47 +871,24 @@ public String delete(String indexName, String type, String id) {
 				: getPersistentEntityFor(clazz).getIndexName();
 		String typeName = hasText(deleteQuery.getType()) ? deleteQuery.getType()
 				: getPersistentEntityFor(clazz).getIndexType();
-		Integer pageSize = deleteQuery.getPageSize() != null ? deleteQuery.getPageSize() : 1000;
-		Long scrollTimeInMillis = deleteQuery.getScrollTimeInMillis() != null ? deleteQuery.getScrollTimeInMillis()
-				: 10000l;
 
-		SearchQuery searchQuery = new NativeSearchQueryBuilder().withQuery(deleteQuery.getQuery()).withIndices(indexName)
-				.withTypes(typeName).withPageable(PageRequest.of(0, pageSize)).build();
+		DeleteByQueryRequest deleteByQueryRequest = new DeleteByQueryRequest(indexName) //
```
Comment (sothawo) [inline src/main/java/org/springframework/data/elasticsearch/core/ElasticsearchRestTemplate.java]: what are these comments at the end of line for? prevent formatting?

File: src/main/java/org/springframework/data/elasticsearch/core/ElasticsearchRestTemplate.java [hunk 2]
```diff
@@ -863,47 +871,24 @@ public String delete(String indexName, String type, String id) {
...
-			try {
-				response = client.bulk(request);
-				checkForBulkUpdateFailure(response);
-			} catch (IOException e) {
-				throw new ElasticsearchException("Error while deleting bulk: " + request.toString(), e);
-			}
+		try {
+			client.deleteByQuery(deleteByQueryRequest, RequestOptions.DEFAULT);
+		} catch (IOException e) {
+			throw new ElasticsearchException("Error for search request with scroll: " + deleteByQueryRequest.toString(), e);
```
Comment (sothawo) [inline src/main/java/org/springframework/data/elasticsearch/core/ElasticsearchRestTemplate.java]: please adjust the error message to the fact that we are deleting and not searching

File: src/main/java/org/springframework/data/elasticsearch/core/ElasticsearchRestTemplate.java (hunk 1 above)
Comment (xhaggi) [reply src/main/java/org/springframework/data/elasticsearch/core/ElasticsearchRestTemplate.java]: Yes.

File: src/main/java/org/springframework/data/elasticsearch/core/ElasticsearchRestTemplate.java (hunk 2 above)
Comment (xhaggi) [reply src/main/java/org/springframework/data/elasticsearch/core/ElasticsearchRestTemplate.java]: Thanks, will change the message

**Comment by xhaggi** [comment]
@sothawo force-pushed with all comments addressed.

**Pull request closed without merging**